charm/src/charm.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Flask Charm entrypoint."""
+
+import logging
+import typing
+
+import ops
+import xiilib.django
+
+logger = logging.getLogger(__name__)
+
+
+class DjangoCharm(xiilib.django.Charm):
+    """Django Charm service."""
+
+    def __init__(self, *args: typing.Any) -> None:
+        """Initialize the instance.
+
+        Args:
+            args: passthrough to CharmBase.
+        """
+        # pylint: disable=useless-parent-delegation
+        super().__init__(*args)
+
+    def gen_env(self) -> dict[str, str]:
+        """Return the environment variables for django scripts.
+
+        Returns:
+           dict with environment variables.
+        """
+        env = super().gen_env()
+        if self._ingress.url:
+            env["DJANGO_BASE_URL"] = self._ingress.url
+        return env
```

Comment:
We might need to think about this more - I'm not yet happy with this interface. It should, at least, probably be the other way around, i.e., you can optionally define a method that returns additional environment variables

Reply:
Sounds reasonable, but that is related to 12 factor not netbox

Reply:
Yes you are right